jitsi-media-transform/src/main/kotlin/org/jitsi/nlj/transform/node/CompliancePcapWriter.kt

```diff
@@ -0,0 +1,263 @@
+import org.jitsi.nlj.rtp.VideoRtpPacket
+import org.jitsi.nlj.util.ReadOnlyStreamInformationStore
+import org.jitsi.utils.MediaType
+import org.jitsi.utils.logging2.Logger
+import org.json.simple.JSONArray
+import org.json.simple.JSONObject
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Paths
+
+class CompliancePcapWriter(
+    private val logger: Logger,
+    private val streamInformationStore: ReadOnlyStreamInformationStore,
+    private val id: String,
+    private val captureEnd: String
+) {
+    private var writer: PcapWriter? = null
+    private val lock = Any()
+    private var contextId: String? = null
+    private var mode: String = CAP_MODE_NONE
+
+    private var captureAudio = false
+    private var captureVideo = false
+
+    fun capId(): String {
+        return "${id}_$captureEnd"
+    }
+
+    fun filename(): String {
+        return "$basePath/${contextId}_${capId()}"
+    }
+
+    fun pcapFilename(): String {
+        return "${filename()}.pcap"
+    }
+
+    fun jsonFilename(): String {
+        return "${filename()}.json"
+    }
+
+    fun validJmtConfig(): Boolean {
+        if (!allowed) {
+            logger.info("Compliance recording:${capId()} is not allowed in jmt.compliance-recording.enabled")
+            return false
+        }
+
+        if (basePath.isEmpty()) {
+            logger.error("Compliance recording:${capId()} jmt.compliance-recording.base-path is not configured")
+            return false
+        }
+
+        if (!Files.isDirectory(Paths.get(basePath))) {
+            logger.error("Compliance recording:${capId()} base-path:$basePath is not a valid directory path")
+            return false
+        }
+
+        return true
+    }
+
+    fun writeMetadata(): Boolean {
+        val meta = JSONObject()
+
+        meta.put("endpoint_id", id)
+        meta.put("context_id", contextId)
+        meta.put("capture_end", captureEnd)
+        meta.put("capture_mode", mode)
+
+        val payloadsMap = JSONArray()
+        streamInformationStore.rtpPayloadTypes.forEach {
+            if ((it.value.mediaType == MediaType.AUDIO && (mode == CAP_MODE_AUDIO || mode == CAP_MODE_AUDIO_VIDEO)) ||
+                (it.value.mediaType == MediaType.VIDEO && (mode == CAP_MODE_VIDEO || mode == CAP_MODE_AUDIO_VIDEO))
+            ) {
+                val entry = JSONObject()
+                entry["payload_type"] = it.value.pt
+                entry["encoding"] = it.value.encoding.toString().lowercase()
+                entry["media_type"] = it.value.mediaType.name.lowercase()
+                entry["clock_rate"] = it.value.clockRate
+                payloadsMap.add(entry)
+            }
+        }
+
+        meta.put("payload_map", payloadsMap)
+
+        // json-simple produces one long flat line
+        // File(jsonFilename()).writeText(meta.toJSONString())
+
+        // gson produces elements on separate lines
+        val gson = GsonBuilder().setPrettyPrinting().create()
```

Comment:
Can we not introduce a new dependency just for this? I think the cleanest way would be to define a kotlin class for your structure and use jackson to serialize to JSON, something like 
```
        data class Metadata(
val endpointId: String,
val contextId: String,
val captureEnd: String,
val payloadType: List<...>
) {
            
        
        fun toJsonString(): String = writer.writeValueAsString(this)
        companon object {
            val writer = jacksonObjectMapper().writerWithDefaultPrettyPrinter()
        }
```

Reply:
Sure we can change this to whatever you guys think is the best, but please note that using jacksonObjectMapper would also introduce new dependency since currently jitsi-media-transform does not include jackson.

Reply:
That's fine, it's already a dependency in jvb/pom.xml

Reply:
Done. Latest commit uses jackson instead of gson.